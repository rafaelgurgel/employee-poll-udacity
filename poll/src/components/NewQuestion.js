import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { handleAddQuestion } from '../actions/shared'
import { useNavigate } from 'react-router-dom'

export default function NewQuestion() {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const [optionOne, setOptionOne] = useState('')
  const [optionTwo, setOptionTwo] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault()
    dispatch(handleAddQuestion(optionOne, optionTwo))
      .then(() => {
        setOptionOne('')
        setOptionTwo('')
        navigate('/home')
      })
  }

  return (
    <div style={{ marginTop: '2rem' }}>
      <h3>Create New Question</h3>
      <form onSubmit={handleSubmit}>
        <label>Would you rather...</label>
        <input
          type="text"
          value={optionOne}
          onChange={(e) => setOptionOne(e.target.value)}
          placeholder="Option One"
          required
          style={{ display: 'block', margin: '0.5rem 0' }}
        />
        <h4 style={{ margin: 0 }}>OR</h4>
        <input
          type="text"
          value={optionTwo}
          onChange={(e) => setOptionTwo(e.target.value)}
          placeholder="Option Two"
          required
          style={{ display: 'block', margin: '0.5rem 0' }}
        />
        <button type="submit">Submit</button>
      </form>
    </div>
  )
}
