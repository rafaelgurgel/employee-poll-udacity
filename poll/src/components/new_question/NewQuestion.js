// src/components/NewQuestion.js
import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { handleAddQuestion } from '../../actions/shared'
import { useNavigate } from 'react-router-dom'
import './new_question.css'

export default function NewQuestion() {
  const dispatch = useDispatch()
  const navigate = useNavigate()
  const [optionOne, setOptionOne] = useState('')
  const [optionTwo, setOptionTwo] = useState('')

  const handleSubmit = (e) => {
    e.preventDefault()
    dispatch(handleAddQuestion(optionOne, optionTwo))
      .then(() => {
        setOptionOne('')
        setOptionTwo('')
        navigate('/home')
      })
  }

  return (
    <div className="new-question-container">
      <h3 className="new-question-title">Create New Question</h3>
      <form className="new-question-form" onSubmit={handleSubmit}>
        <label>Would you rather...</label>
        <input
          type="text"
          value={optionOne}
          onChange={(e) => setOptionOne(e.target.value)}
          placeholder="Option One"
          required
          className="new-question-input"
        />

        <h4 className="new-question-or">OR</h4>

        <input
          type="text"
          value={optionTwo}
          onChange={(e) => setOptionTwo(e.target.value)}
          placeholder="Option Two"
          required
          className="new-question-input"
        />

        <button type="submit" className="new-question-submit">
          Submit
        </button>
      </form>
    </div>
  )
}
