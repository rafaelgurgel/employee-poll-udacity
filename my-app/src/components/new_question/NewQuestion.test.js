import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { BrowserRouter } from 'react-router-dom'
import NewQuestion from './NewQuestion'
import '@testing-library/jest-dom'

const mockReducer = (state = {}) => state
const mockStore = createStore(mockReducer)

describe('NewQuestion Component', () => {
  beforeAll(() => {
    jest.spyOn(console, 'warn').mockImplementation((message) => {
      if (!message.includes('React.startTransition')) {
        console.warn(message)
      }
    })
  })

  afterAll(() => {
    jest.restoreAllMocks()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Provider store={mockStore}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
