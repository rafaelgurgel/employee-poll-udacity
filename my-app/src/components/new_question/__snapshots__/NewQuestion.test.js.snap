// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="new-question-container"
  >
    <h3
      class="new-question-title"
    >
      Create New Question
    </h3>
    <form
      class="new-question-form"
    >
      <label>
        Would you rather...
      </label>
      <input
        class="new-question-input"
        placeholder="Option One"
        required=""
        type="text"
        value=""
      />
      <h4
        class="new-question-or"
      >
        OR
      </h4>
      <input
        class="new-question-input"
        placeholder="Option Two"
        required=""
        type="text"
        value=""
      />
      <button
        class="new-question-submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
